### PR TITLE
fix(ci): update regtest with default wallet and generate blocks

### PR DESCRIPTION
**Pull request by vdovhanych**
This PR addresses #102 

This seems to be the easiest solution to running the backend and filling it with bitcoins. As we need to wait a bit for the service to start, I had to add sleep, and in general, generating bitcoins will slow down the startup of the blockbook (approx 10-15 seconds) not sure if this could cause some problems with the tenv server @grdddj

### CODE REVIEW

**Review by tsusanka — COMMENTED** [review]
So tell me again why can't we use just `bin/bitcoin-cli` since we have `cd` into the folder?

Other then that I am assigning @grdddj and he'll check if this does the trick for him :).

**Comment by vdovhanych** [comment]
> So tell me again why can't we use just `bin/bitcoin-cli` since we have `cd` into the folder?
> 
> 
> 
> Other then that I am assigning @grdddj and he'll check if this does the trick for him :).

Will shorten it, you're right it will work 😁

**Comment by vdovhanych** [comment]
Force pushing shortened commands and changed amount of blocks generated so it's a bit faster to load.

**Review by tsusanka — APPROVED** [review]
LGTM but as mentioned let's wait for @grdddj

File: docker/bitcoin-regtest/entrypoint.sh
Comment (grdddj) [inline docker/bitcoin-regtest/entrypoint.sh:14]: I would say that `-generate 150` is enough here, we do not need so many BTC (1024 blocks will yield 14700 BTC, 150 is the "halving" interval, and we get 2500 spendable BTC from it - more than enough).

The second argument is also optional, based on [docs](https://developer.bitcoin.org/reference/rpc/generate.html), we do not need to include it (unless there is a reason for it - @vdovhanych ?)

File: docker/bitcoin-regtest/entrypoint.sh
Comment (grdddj) [inline docker/bitcoin-regtest/entrypoint.sh:8]: Here I am confirming that changing the config path to relative is really not possible, even though it is in the same dir as we `cd'd`

File: docker/bitcoin-regtest/entrypoint.sh
Comment (grdddj) [inline docker/bitcoin-regtest/entrypoint.sh:20]: In this command, I was able to replace the absolute path with relative - deleting all the `/opt/coins/blockbook/bitcoin_regtest` prefixes of locations

**Comment by grdddj** [comment]
>  generating bitcoins will slow down the startup of the blockbook (approx 10-15 seconds) not sure if this could cause some problems with the tenv server @grdddj

It is not so slow on my machine, I think it is alright. I did not try run both in containers with docker-compose yet, but I imagine the app must wait until both containers to initialize.

As for this PR, I am approving that. I made some smaller changes and additions to it in https://github.com/trezor/trezor-user-env/pull/100 (based on my comments here), so that the combination of `controller` and `regtest image` is there. We could migrate these `entrypoint` changes into this PR to make it separate from `controller`

**Review by grdddj — APPROVED** [review]

**Comment by vdovhanych** [comment]
Yeah let's move it here or just merge this before your's and that should also be fine i guess no ?

**Comment by grdddj** [comment]
> Yeah let's move it here or just merge this before your's and that should also be fine i guess no ?

Agreed, feel free to transfer the `entrypoint` changes here and merge, so my PR only touches `controller`

**Comment by vdovhanych** [comment]
> > Yeah let's move it here or just merge this before your's and that should also be fine i guess no ?
> 
> Agreed, feel free to transfer the `entrypoint` changes here and merge, so my PR only touches `controller`

Ok

**Comment by vdovhanych** [comment]
Force pushed with the changes you suggested. should be ready to go now.

**Comment by grdddj** [comment]
> Force pushed with the changes you suggested. should be ready to go now.

Good, thanks for that. I agree we are ready to merge - my PR can follow (after review and green light)

File: docker/bitcoin-regtest/entrypoint.sh
Comment (grdddj) [inline docker/bitcoin-regtest/entrypoint.sh:21]: Maybe last thing for discussion - what about splitting these long commands into multiple lines, like
```sh
bin/blockbook \
    -blockchaincfg=config/blockchaincfg.json \
    -datadir=/opt/coins/data/bitcoin_regtest/blockbook/db \
    -sync \
    -internal=:19021 \
    -public=:19121 \
    -certfile=cert/blockbook \
    -explorer= \
    -log_dir=logs
```

Actually seeing this uncovered to me that `-explorer` is empty is it alright?

Comment (vdovhanych) [reply docker/bitcoin-regtest/entrypoint.sh:21]: Yes tahat should be Ok, all of the packages are run this way. Don't really think it's necessary to modify the commands any further I'm fine with how it is know but in all honesty whatever you guys like more I will be fine with.

**Comment by vdovhanych** [comment]
Had to update the entrypoint again as it wasn't creating the wallet and blocks. Set it to sleep 5 and now it is consistent with creating that so now it should be ok.

File: docker/bitcoin-regtest/entrypoint.sh
Comment (tsusanka) [reply docker/bitcoin-regtest/entrypoint.sh:21]: Yeah that looks nicer. So @vdovhanych let's do this final minor change and ship it?

Comment (vdovhanych) [reply docker/bitcoin-regtest/entrypoint.sh]: Done, will test the docker image after it's built and if it's ok i will merge it then.

Comment (grdddj) [reply docker/bitcoin-regtest/entrypoint.sh]: I agree with merging

Comment (tsusanka) [reply docker/bitcoin-regtest/entrypoint.sh]: Yes, do it!

**Comment by vdovhanych** [comment]
It works fine, merging it.

**Pull request merged**
